---
title: "Index"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Project structure

We propose this structure:

* DATA: divided by project
* SOFTWARE: common to all projects
* REFERENCES (genome, transcriptome, etc)

```{bash mkdir, eval=FALSE}
mkdir data software references data/SRAtest
```

## References 
+  [Getting references](references.html)

## Software installation in AWS

+  [Software standalone](software_minimal.html)
+  [Nextflow](nextflow.html)
+  [Nextflow batch](nextflow_batch.html)
+  [AWS CLI](awsclli.html)

## Software commands

+  [STAR](star.html)
+  [SALMON](salmon.html)

# Nextflow

## Pipelines (using nextflow)

+  [FETCH from SRA](sraFetch.html)
+  [RNASEQ](rnaseq.html)

# Dataset description

+  [Dataset](description.html)
