---
title: "Requiered software for the analysis of AS using RNA-Seq"
author: "Estefania Mancini"
date: "2022-10-17"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
```

# Project structure
We propose this structure:

* DATA: divided by project
* SOFTWARE: common to all projects
* REFERECES (genome, transcriptome, etc)

```{bash mkdir, eval=FALSE}
mkdir data software references data/SRAtest
```

# Software

```{bash cd, eval=FALSE}
cd software/
```

## sratoolkit

source: https://github.com/ncbi/sra-tools

```{bash getsratoolkit, eval=FALSE}

wget https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-centos_linux64.tar.gz
tar -zxf sratoolkit.current-centos_linux64.tar.gz 
export PATH=$PATH:$PWD/sratoolkit.3.0.0-centos_linux64/bin
```

### Configure vdb

```{bash configure, eval=FALSE}
vdb-config --interactive
```

Source: https://github.com/ncbi/sra-tools/wiki/05.-Toolkit-Configuration

### Get some test data from dataset

+ Option 1
```{bash dump1, eval=FALSE}
cd  ~/data/SRAtest
fastq-dump --stdout -X 2 SRR390728

```

+ Option 2

```{bash fetch, eval=FALSE}
prefetch SRR17333143
fastq-dump --outdir fastq --gzip --skip-technical  --readids --read-filter pass --dumpbase --split-3 --clip SRR17333143 &
```


## Fastqc

### Installation
source https://www.bioinformatics.babraham.ac.uk/projects/fastqc/

* We should install JAVA first. Any clue? 

```{bash java, eval=FALSE}
sudo yum install java-1.8.0-openjdk-devel
```

Then we get fastqc:

```{bash fastqc, eval=FALSE}
wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip
cd FastQC
chmod ug+x fastqc
export PATH=$PATH:/home/ec2-user/software/FastQC/
```

 * Run

```{bash fastqcRun, eval=FALSE}
cd ~/data/SRAtest
fastqc  sample.fq
```
 
# MultiQC

From here we can use conda! 
Build a report to visualize raw data

To install MultiQC, simply run pip install multiqc on the command line.
If you use conda, you can run conda install -c bioconda multiqc instead.
See the installation instructions for more help.

```{bash pipInstall, eval=FALSE}
sudo yum install epel-release
sudo amazon-linux-extras install epel
sudo yum install python-pip
pip3 install multiqc
```

Finally run MultiQC on our samples:

```{bash multiQC, eval=FALSE}
cd ~/data/SRAtest
multiqc -p dat 
```

# Alignments

## Download references
From  Gencode v39:  https://www.gencodegenes.org/human/release_39.html

* GTF superset: 
```{bash gtf, eval=FALSE}
cd  ~/references
wget https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_39/gencode.v39.chr_patch_hapl_scaff.annotation.gtf.gz .
```

Total wall clock time: 1m 7s (66 sec)
Downloaded: 1 files, 49M in 1m 6s (751 KB/s)

* Fasta Transcriptome:

```{bash transcriptome, eval=FALSE}
wget https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_39/gencode.v39.transcripts.fa.gz
```
45 sec


* Fasta genome: 

```{bash genome, eval=FALSE}
wget https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_39/GRCh38.p13.genome.fa.gz .
```
Total wall clock time: 18m 43s
Downloaded: 1 files, 848M in 18m 42s (774 KB/s)

## STAR

https://github.com/alexdobin/STAR
https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf

* Installation:

* Get latest STAR source from releases

```{bash STAR, eval=FALSE}
wget https://github.com/alexdobin/STAR/archive/2.7.10a.tar.gz
```
2 sec

```{bash tar, eval=FALSE}
tar xvzf 2.7.10a.tar.gz 
```

* Compile


```{bash zlib, eval=FALSE}
sudo yum install zlib
sudo yum install  install zlib1g-dev
```

```{bash gcc, eval=FALSE}
sudo yum -y install gcc
sudo  yum install gcc-c++ zlib1g-dev

```
Installed: gcc.x86_64 0:7.3.1-15.amzn2    
  

```{bash compileSTAR2, eval=FALSE}
cd source
make STAR
./STAR --help

sudo ln -s /home/ec2-user/software/STAR-2.7.10a/source/STAR /usr/local/bin/STAR
```
Done! 

* Make an index for 75pb, basic options:

--runThreadN NumberOfThreads

--runMode genomeGenerate

--genomeDir /path/to/genomeDir: This directory has to be created (with mkdir) before STAR run and needs to have writing permissions.

--genomeFastaFiles /path/to/genome/fasta1: specifies one or more FASTA files with the genome reference sequences.


--sjdbGTFfile /path/to/annotations.gtf

--sjdbOverhang ReadLength-1

```{bash genomeIndex, eval=FALSE}
STAR --runThreadN 4 --runMode genomeGenerate --genomeDir /home/ec2-user/references/GRCh38_STAR --genomeFastaFiles /home/ec2-user/references/GRCh38.p13.genome.fa --sjdbGTFfile /home/ec2-user/references/gencode.v39.chr_patch_hapl_scaff.annotation.gtf --sjdbOverhang 74 
```

* Por que ENCODE and not ENSEMBL?
* Me quede sin memoria
 
# Gene quantification

We use STAR default output for gene count

# Transcript quantification

## Salmon  (favourite)

* Installation
https://docs.anaconda.com/anaconda/install/linux/
https://anaconda.org/bioconda/salmon

```{bash anacondareq, eval=FALSE}
sudo yum install libXcomposite libXcursor libXi libXtst libXrandr alsa-lib mesa-libEGL libXdamage mesa-libGL libXScrnSaver
```

```{bash getAnaconda, eval=FALSE}
wget https://repo.anaconda.com/archive/Anaconda3-2022.05-Linux-x86_64.sh .
```

Then:

```{bash installSalmon, eval=FALSE}
conda install -c bioconda salmon
conda install -c "bioconda/label/cf201901" salmon
conda config --add channels bioconda
conda create -n salmon salmon
conda activate salmon
salmon -h
```



## RSEM    (2nd favourite)

https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-323
https://github.com/deweylab/RSEM

* Installation
https://anaconda.org/bioconda/rsem

```{bash Rsem, eval=FALSE}
conda install -c bioconda rsem
conda install -c "bioconda/label/cf201901" rsem
```

I went out of space :/

# Exon/intron quantification

## ASpli


* Install R latest version from source

https://docs.rstudio.com/resources/install-r-source/

```{bash Rsource, eval=FALSE}
sudo yum-builddep R
```

Check which is last version:
https://cran.rstudio.com/src/base/

Today is   R-4.2.1.tar.gz          

```{bash Rversion, eval=FALSE}
export R_VERSION=4.2.1
```

```{bash RversionVar, eval=FALSE}
curl -O https://cran.rstudio.com/src/base/R-4/R-${R_VERSION}.tar.gz
#0 sec
tar -xzvf R-${R_VERSION}.tar.gz
cd R-${R_VERSION}
```

* Install

```{bash Rconf, eval=FALSE}

./configure \
    --prefix=/opt/R/${R_VERSION} \
    --enable-memory-profiling \
    --enable-R-shlib \
    --with-blas \
    --with-lapack
```


* Build and install R

Build and install R by running the following commands:

```{bash Rmake, eval=FALSE}
make
sudo make install
```

* Verify R installation
Test that R was successfully installed by running:


```{bash Rversionsource, eval=FALSE}
/opt/R/${R_VERSION}/bin/R --version
```

* Create a symlink to R
To ensure that R is available on the default system PATH variable, create symbolic links to the version of R that you installed:

```{bash symlink, eval=FALSE}
sudo ln -s /opt/R/${R_VERSION}/bin/R /usr/local/bin/R
sudo ln -s /opt/R/${R_VERSION}/bin/Rscript /usr/local/bin/Rscript
```


* (Optional) Install recommended packages
We recommend installing several optional system dependencies that are used by common R packages. Additional information about installing them is provided in our documentation.



*  Bioconductor

```{R bioc, eval=FALSE}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install(version = "3.15")
```


```{bash xml2, eval=FALSE}
sudo yum install libxml2-devel
yum install make gcc perl pcre-devel zlib-devel
wget https://ftp.openssl.org/source/openssl-3.0.5.tar.gz

sudo yum install perl-IPC-Cmd

tar xvf openssl-3.0.5.tar.gz 
cd 
./config --prefix=/usr --openssldir=/etc/ssl --libdir=lib no-shared zlib-dynamic

export LD_LIBRARY_PATH=/usr/local/lib:/usr/local/lib64
echo "export LD_LIBRARY_PATH=/usr/local/lib:/usr/local/lib64" >> ~/.bashrc
openssl version
```

OpenSSL 3.0.5 5 Jul 2022 (Library: OpenSSL 3.0.5 5 Jul 2022)

* Useful packages from Bioconductor

```{R ASpli, eval=FALSE}
BiocManager::install("biomaRt")
BiocManager::install("rtracklayer")
BiocManager::install("AnnotationDbi")
BiocManager::install("GenomicFeatures")
BiocManager::install("ensembldb")
BiocManager::install("BSgenome")
BiocManager::install("BSgenome")
BiocManager::install("biovizBase")
BiocManager::install("DESeq2") 
```


* ASpli

```{R ASpli0, eval=FALSE}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("ASpli")
```

* Other useful packages:
```{R packages, eval=FALSE}
install.packages("ggplot2")
install.packages("rmarkdown")
```

